### ISSUE TITLE
Possible to generate GS1 codes?

### ISSUE THREAD

**Issue by MarcelNehring** [zxing-cpp/zxing-cpp#374]
First of all thank you for providing this library. 👍 

I would like to know, if it is possible to generate GS1 codes with zxing-cpp? I am especially interested in generating datamatrix codes with valid GS1 content. However, I don't seem to be able to generate a valid GS1 datamatrix. Since I don't know what the correct way to encode a FNC1 character is I am not sure whether this is the problem I am facing or if I am doing wrong something else.
I have tried using codeword 232 when passing my string to the `encode` method of my writer but the resulting image seems to be invalid.

Any help or hint on how to correctly encode GS1 data would be appreciated

**Comment by axxel** [comment]
I know very little about the encoding part of the code base which has not been touched much since it was ported from Java. Before I even look at if the code even supports that, have you considered using [zint](https://sourceforge.net/projects/zint/) to generate your symbols? It is a much more feature complete alternative. See also #332.

**Comment by axxel** [comment]
> I saw that there is #332 so not sure if you consider it worth the effort at this time?

I most definitively don't think it is worth putting work into that part of the library as long as a `zint` integration/collaboration is on the table. May I ask why simply using zint in your project was not an option?

**Comment by MarcelNehring** [comment]
`zxing-cpp` is available on vcpkg and thus it is very easy to integrate it into our build process. Even creating a port-overlay to use older version 1.1.1 (last version not requiring C++17) and making it compatible to VS2015 was very easy. `zint` doesn't seem to have a vcpkg package available.
Additionally the API of `zxing-cpp` looks much more modern and easier to use than `zint` (we don't generate images but rather use the bit matrix and draw the images ourselves). I wasn't aware of `zint` until you mentioned it here and I had already integrated it into our code.

So I feared that switching to `zint` would take me some time and therefore wanted to spend a bit more time trying to figure out whether `zxing-cpp` can handle GS1. When I had figured out it can't, it wasn't much more effort to fix that (at least for the tests I did).
Would it have been more complex ore take me too much time I would have stopped trying and probably would have switched to `zint`.

**Comment by axxel** [comment]
Thanks for that detailed feedback to my question. The conclusion I draw from that is: there is real value in providing some kind of c++ front-end/wrapping for zint as part of zxing-cpp with a similarly simple API as `ReadBarcode`. That would neither necessarily nor likely be as powerful and feature complete as the native zint API is but therefore be more accessible, both in terms of packaging and available wrappers. (Mentioning @gitlost here, in case he is not following this conversation but might want to draw his own/different conclusions).

Am I right that the solution to your particular issue is the local workaround that you found?

I'd like to point out that there seems to be an issue with the DataMatrix encoder when it comes to symbols of size 144x144: https://github.com/Touchless-ID/zxing-cpp/commit/b3b8025aef506fc6f295195ec6592d4846ea2bbb. My advise: avoid those if possible.

Last bit of advice: UPGRADE your build setup! ;)

**Comment by gitlost** [comment]
Yes Zint has poor distribution and infrastructure support currently. Though there are so many package managers out there now it's hard to keep track, and one is very dependent on someone being willing to maintain them.

I hadn't thought a C++ interface to zint was necessary, as it's trivial to call from C++, but maybe I'm wrong. What is not in doubt is the benefits to zint of some form of integration with zxing-cpp, as mentioned in #332.

**Comment by axxel** [comment]
Reopened to lower the chance of this being reported again as it is still an open issue, that will hopefully be fixed with #332.
